### PR TITLE
Login and projects api

### PR DESCRIPTION
**Pull request by YahorKazlou**
- add Prettier and Eslint checks to GitHub actions on pull requests
- add unit testing library and debuging for curent file
- add tests for login action
- add jest debuging

### CODE REVIEW

File: src/components/Projects/SearchBar.jsx [hunk 1]
```diff
@@ -17,7 +17,4 @@ const SearchBar = ({ onSearch }) => {
 
 export default SearchBar;
 
-const oneF = () => {
-
-};
-
+const oneF = () => {};
```
Comment (Den4ik5) [inline src/components/Projects/SearchBar.jsx]: do we need this?

File: src/components/Projects/index.jsx [hunk 2]
```diff
@@ -1,66 +1,24 @@
...
-        link: '',
-    },
-    {
-        name: 'Spring Scurity',
-        text: 'Protects your application with comprehensive and extensible authentication and authorization support.',
-        imgUrl: springSecurity,
-        link: '',
-    },
-];
+    const handleApi = async (searchTerm) => {
```
Comment (Den4ik5) [inline src/components/Projects/index.jsx]: bad naming, don't add Api to the methods.

File: src/api/projects.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { apiHandler } from './apiHandler';
+
+export const getProjectsApi = (searchTerm) => {
```
Comment (Den4ik5) [inline src/api/projects.js]: no need for Api methods must describe what is happening so getProjects is enough

File: src/api/auth.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { apiHandler } from './apiHandler';
+
+export const loginApi = (login, password) =>
```
Comment (Den4ik5) [inline src/api/auth.js]: just login

File: src/api/apiHandler.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+export const apiHandler = async (endpoint, method = 'GET', requestBody) => {
```
Comment (Den4ik5) [inline src/api/apiHandler.js:1]: you should also handle errors here probably

File: src/components/Projects/index.jsx [hunk 6]
```diff
@@ -1,66 +1,24 @@
...
+        const response = await getProjectsApi(searchTerm);
+        const data = await response.json();
+        if (response.status === 200) setProjects(data.data);
+    };
 
-const Projects = () => {
-    const [projects, setProjects] = useState(defaultProjects);
+    useEffect(() => {
+        handleApi();
+    }, []);
```
Comment (Den4ik5) [inline src/components/Projects/index.jsx]: save projects to the store

File: src/hooks/useWindowDimensions.jsx [hunk 7]
```diff
@@ -1,26 +1,26 @@
-import { useState, useEffect } from "react";
+import { useState, useEffect } from 'react';
 
 function getWindowDimensions() {
-      const { innerWidth: width, innerHeight: height } = window;
-      return {
-            width,
```
Comment (Den4ik5) [inline src/hooks/useWindowDimensions.jsx:6]: your linter is broken?

File: src/tests/user-actions.test.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import { loginApi } from '../api';
+import { enableFetchMocks } from 'jest-fetch-mock';
+
+enableFetchMocks();
+
+describe('userActions', () => {
+    beforeEach(() => {
+        fetch.resetMocks();
+    });
+    it('Login successful, creates LOGIN_PENDING and LOGIN_SUCCESS', async () => {
```
Comment (Den4ik5) [inline src/tests/user-actions.test.js]: usually tests start with 
it('Should do/return/etc ...

**Review by Den4ik5 — CHANGES_REQUESTED** [review]

File: src/hooks/useWindowDimensions.jsx (hunk 7 above)
Comment (YahorKazlou) [reply src/hooks/useWindowDimensions.jsx:6]: No. Linter is not broken. These are its default settings.

File: src/components/Projects/SearchBar.jsx (hunk 1 above)
Comment (YahorKazlou) [reply src/components/Projects/SearchBar.jsx]: done

File: src/components/Projects/index.jsx (hunk 2 above)
Comment (YahorKazlou) [reply src/components/Projects/index.jsx]: done

File: src/api/projects.js (hunk 3 above)
Comment (YahorKazlou) [reply src/api/projects.js]: done

File: src/api/auth.js (hunk 4 above)
Comment (YahorKazlou) [reply src/api/auth.js]: done

File: src/api/apiHandler.js (hunk 5 above)
Comment (YahorKazlou) [reply src/api/apiHandler.js:1]: done

File: src/components/Projects/index.jsx (hunk 6 above)
Comment (YahorKazlou) [reply src/components/Projects/index.jsx]: done

File: src/tests/user-actions.test.js (hunk 8 above)
Comment (YahorKazlou) [reply src/tests/user-actions.test.js]: done

**Pull request merged**